test/integration/shell_int_test.go

```diff
@@ -292,27 +291,33 @@ func (suite *ShellIntegrationTestSuite) TestNestedShellNotification() {
 	cfg, err := config.New()
 	suite.Require().NoError(err)
 
+	// Setup subshell. Since our CI tests are running in bash, override with zsh on macOS.
 	var ss subshell.SubShell
 	env := []string{"ACTIVESTATE_CLI_DISABLE_RUNTIME=false"}
 	if runtime.GOOS != "darwin" || !condition.OnCI() {
 		ss = subshell.New(cfg)
 	} else {
-		os.Setenv("SHELL", "zsh") // GitHub actions runs on bash, so override with zsh
+		cfg.Set(subshell.ConfigKeyShell, "zsh")
 		ss = subshell.New(cfg)
-		os.Unsetenv("SHELL")
+		cfg.Set(subshell.ConfigKeyShell, "")
 		env = append(env, "SHELL=zsh")
 	}
-	os.Setenv(constants.HomeEnvVarName, ts.Dirs.HomeDir)
-	err = subshell.ConfigureAvailableShells(ss, cfg, nil, sscommon.InstallID, true) // mimic installer
-	os.Unsetenv(constants.HomeEnvVarName)
-	suite.Require().NoError(err)
 
-	os.Setenv(constants.HomeEnvVarName, ts.Dirs.HomeDir)
-	rcFile, err := ss.RcFile()
-	os.Unsetenv(constants.HomeEnvVarName)
-	suite.Require().NoError(err)
+	// Setup shell-specific RC file in test-specific home directory.
+	var rcFile, rcFileAppend string
+	switch ss.Shell() {
+	case "bash":
+		rcFile = filepath.Join(ts.Dirs.HomeDir, ".bashrc")
+		rcFileAppend = "bashrc_append.sh"
+	case "zsh":
+		rcFile = filepath.Join(ts.Dirs.HomeDir, ".zshrc")
+		rcFileAppend = "zshrc_append.sh"
+	default:
+		suite.Fail("Unknown subshell: %s", ss.Shell())
+	}
 	suite.Require().Equal(filepath.Dir(rcFile), ts.Dirs.HomeDir, "rc file not in test suite homedir")
-	suite.Require().FileExists(rcFile)
+	fileutils.TouchFileUnlessExists(rcFile)
+	sscommon.WriteRcFile(rcFileAppend, rcFile, sscommon.InstallID, nil)
```

Comment:
I don't follow why we need this? The `TouchFile` above seems to already be addressing the needs of this test. And without passing any env information this wouldn't actually be writing any templated data. The InstallID is also irrelevant because you're using the append template, which doesn't use this.

Pretty sure you can drop this and your test would still run as expected.

Reply:
Ahhh that's the part I was missing. I was looking at my local zshrc_append file and wondering what you were looking to set up -.-